app/src/main/java/io/github/jbellis/brokk/analyzer/cpp/NamespaceProcessor.java

```diff
@@ -0,0 +1,272 @@
+        }
+        return skeletons;
+    }
+
+    private String extractEnumSkeletonFromNode(TSNode enumNode, String fileContent) {
+        var nameNode = enumNode.getChildByFieldName("name");
+        String enumName = "";
+        if (nameNode != null && !nameNode.isNull()) {
+            enumName = ASTTraversalUtils.extractNodeText(nameNode, fileContent);
+        }
+
+        boolean isScopedEnum = false;
+        var enumText = ASTTraversalUtils.extractNodeText(enumNode, fileContent);
+        if (enumText.startsWith("enum class")) {
+            isScopedEnum = true;
+        }
+
+        var skeleton = new StringBuilder(256);
+        if (isScopedEnum) {
+            skeleton.append("enum class ").append(enumName).append(" {\n");
+        } else {
+            skeleton.append("enum ").append(enumName).append(" {\n");
+        }
+
+        var bodyNode = enumNode.getChildByFieldName("body");
+        if (bodyNode != null && !bodyNode.isNull()) {
+            var enumValues = new ArrayList<String>();
+
+            for (int i = 0; i < bodyNode.getChildCount(); i++) {
+                var child = bodyNode.getChild(i);
+                if (child != null && !child.isNull() && "enumerator".equals(child.getType())) {
+                    var enumeratorNameNode = child.getChildByFieldName("name");
+                    if (enumeratorNameNode != null && !enumeratorNameNode.isNull()) {
+                        var enumValueName = ASTTraversalUtils.extractNodeText(enumeratorNameNode, fileContent);
+                        if (!enumValueName.isEmpty()) {
+                            enumValues.add(enumValueName);
+                        }
+                    }
+                }
+            }
+
+            for (int i = 0; i < enumValues.size(); i++) {
+                skeleton.append("        ").append(enumValues.get(i));
+                if (i < enumValues.size() - 1) {
+                    skeleton.append(",");
+                }
+                skeleton.append("\n");
+            }
+        }
+
+        skeleton.append("    }");
+        return skeleton.toString();
+    }
+
+    private String extractFunctionSignature(TSNode funcNode, String fileContent) {
+        var declarator = funcNode.getChildByFieldName("declarator");
+        if (declarator == null || declarator.isNull()) {
+            return "";
+        }
+
+        String returnType = "";
+        var typeNode = funcNode.getChildByFieldName("type");
+        if (typeNode != null && !typeNode.isNull()) {
+            returnType = ASTTraversalUtils.extractNodeText(typeNode, fileContent) + " ";
+        }
+
+        String signature = returnType + ASTTraversalUtils.extractNodeText(declarator, fileContent);
+
+        for (int i = 0; i < funcNode.getChildCount(); i++) {
+            var child = funcNode.getChild(i);
+            if (child == null || child.isNull()) continue;
+
+            String childType = child.getType();
+            if ("noexcept_specifier".equals(childType) ||
```

Comment:
I wonder if all these TS types should live in some enum or class with static string fields?